qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -89,22 +104,24 @@ def _compute_gradients(
 
         Returns:
             List of pairs consisting of gradient and excitation operator.
+            parameters = ParameterVector("θ", self._num_parameters)
+            times = ParameterVector("t", self.reps * len(self.operators))
         """
         res = []
-        # compute gradients for all excitation in operator pool
         for exc in self._excitation_pool:
-            # add next excitation to ansatz
             self._ansatz.operators = self._excitation_list + [exc]
-            # set the current ansatz
-            vqe.ansatz = self._ansatz
-            # evaluate energies
-            parameter_sets = theta + [-self._delta] + theta + [self._delta]
-            energy_evaluation = vqe.get_energy_evaluation(self._main_operator)
-            energy_results = energy_evaluation(np.asarray(parameter_sets))
-            # compute gradient
-            gradient = (energy_results[0] - energy_results[1]) / (2 * self._delta)
-            res.append((np.abs(gradient), exc))
-
+            if self.gradient._grad_method.analytic:
+                vqe.ansatz = self._ansatz
+            else:
+                vqe.ansatz = self._ansatz.decompose()
+            param_sets = vqe._ansatz_params
+            op = vqe.construct_expectation(theta, self._main_operator)
+            state_grad = self.gradient.convert(operator=op, params=param_sets)
+            # Assign the parameters and evaluate the gradient
+            value_dict = {param_sets[-1]: 0.0}
+            state_grad_result = state_grad.assign_parameters(value_dict).eval()
+            print("State gradient computed with parameter shift", state_grad_result)
```

Comment:
```suggestion
            logger.info("State gradient computed with parameter shift", state_grad_result)
```
Instead of always printing this, we should use the `logger` instead such that a user can configure whether or not they would like to see this kind of output.